scripts/cross-ghc

```diff
@@ -0,0 +1,305 @@
+       >&2 echo "Nothing found at $llvmPath, which was specificed as the path to LLVM5! Please make sure LLVM is installed."
+       exit 1
+    fi
+  elif [ -d "$defaultLlvmPath" ]; then
+    llvmPath=$defaultLlvmPath
+  else
+     >&2 echo "Could not find LLVM5 at $defaultLlvmPath, and an alternative path was not defined in \$ghcCrossLlvmPath."
+    exit 1
+  fi
+}
+
+# Set the path to the LLVM5.
+function ensureLibffiPath() {
+  if [ -n "$ghcCrossLibffiPath" ]; then
+    libffiPath="$ghcCrossLibffiPath"
+    if [ ! -d "$llvmPath" ]; then
+       >&2 echo "Nothing found at $libffiPath, which was specificed as the path to libffi! Please make sure libffi is installed."
+       exit 1
+    fi
+  elif [ -d "$defaultLibffiPath" ]; then
+    libffiPath=$defaultLibffiPath
+  else
+     >&2 echo "Could not find libffi at $defaultLibffiPath, and an alternative path was not defined in \$ghcCrossLibffiPath."
+    exit 1
+  fi
+}
+
+function ensureGhcOnPath() {
+  if [ ! $(which $targetPrefix-ghc) ]; then
+    >&2 echo "Could not find $targetPrefix-ghc on path! Are you sure you have"
+    >&2 echo "built the GHC binaries for this distribution? See 'cross-ghc help'"
+    >&2 echo "for more information."
+    >&2 echo ""
+    exit 1
+  fi
+}
+
+
+# Build GHC binaries for a specific target.
+function buildGhcBinaries() {
+  echo "Starting process to build the GHC for the target platform."
+  echo ""
+  # Make sure the distribution exists and then configure/build it.
+  getBinaryDistruibution \
+    && (cd $crossGhcDistributionPath/ghc-$targetPrefix && buildAndConfigureDistribution)
+}
+
+function getBinaryDistruibution() {
+  if [ ! -d "$crossGhcDistributionPath/ghc-$targetPrefix" ]; then
+    # Download the distribution, unzip and untar it, and then finally configure
+    # and build the GHC binaries.
+    echo "Choosing distribution: $ghcDistribution"
+    echo ""
+    mkdir -p $crossGhcDistributionPath \
+      && curl -o $crossGhcDistributionPath/ghc-$targetPrefix.tar.xz \
+         -O -L -C - $ghcDistribution \
+         --retry 20 \
+      && gunzip -c $crossGhcDistributionPath/ghc-$targetPrefix.tar.xz > $crossGhcDistributionPath/ghc-$targetPrefix.tar \
+      && tar xf $crossGhcDistributionPath/ghc-$targetPrefix.tar -C $crossGhcDistributionPath \
+      && mv $crossGhcDistributionPath/$ghcVersion $crossGhcDistributionPath/ghc-$targetPrefix \
+      && echo "Downloaded and unpacked distribution to $crossGhcDistributionPath/ghc-$targetPrefix" \
+      && echo ""
+  else
+    echo "Using existing GHC distribution for $target."
+    echo ""
+  fi
+}
+
+function buildAndConfigureDistribution() {
+  export PATH="$toolchainWrapperPath:$llvmPath:$PATH"
+  ./configure \
+    --prefix=$crossGhcDistributionPath/ghc-$targetPrefix \
+    --target=$targetPrefix \
+    --host=$hostConfigure \
+    --build=$hostConfigure \
+    && make install \
+    && echo "" \
+    && echo "Finished building $targetPrefix-ghc." \
+    && echo ""
+}
```

Comment:
I've been confused about the naming here. We don't actually `build` anything. `make install` just copies the binary distribution into place. Not sure if we should call this `build`?

Reply:
I suppose `setup` or something like that would make more sense?

Reply:
I've renamed the command to `setup-ghc` now, which is a bit more fitting probably.